runtime/vm/BytecodeInterpreter.hpp

```diff
@@ -4021,13 +4021,46 @@ class INTERPRETER_CLASS
 		return EXECUTE_BYTECODE;
 	}
 
-	/* ToDo: unimplemented - https://github.com/eclipse-openj9/openj9/issues/13614 */
 	/* jdk.internal.misc.Unsafe: public native boolean isFlattenedArray(Class<?> clz); */
 	VMINLINE VM_BytecodeAction
 	inlUnsafeIsFlattenedArray(REGISTER_ARGS_LIST)
 	{
+		j9object_t cls = *(j9object_t*)_sp;
+
 		updateVMStruct(REGISTER_ARGS);
-		Assert_VM_unreachable();
+		buildInternalNativeStackFrame(REGISTER_ARGS);
+
+		I_32 result = (I_32)FALSE;
+		if (NULL != cls) {
+			J9Class *fieldClazz = J9VM_J9CLASS_FROM_HEAPCLASS(_currentThread, cls);
+			result = (I_32)J9_IS_J9CLASS_FLATTENED(fieldClazz);
```

Comment:
What is the expected behaviour if `cls` is not an array class?

Reply:
If cls isn't an array class then it should return false. I wasn't doing it before, but I fixed it in 25faee9.
I also noticed that the expected behavior of isFlattened is to throw a NullPointerException when its parameter is null (instead of just returning false), which I also fixed in 25faee9.